File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20034.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.FoodIngredientsMapProducts
import scala.collection.immutable.Seq

/**
 * 
 * @param original 
 * @param originalName 
 * @param ingredientImage 
 * @param meta 
 * @param products 
 */
case class InlineResponse20034(original: String,
                originalName: String,
                ingredientImage: String,
                meta: Seq[String],
                products: Seq[FoodIngredientsMapProducts]
                )

object InlineResponse20034 {
    /**
     * Creates the codec for converting InlineResponse20034 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20034] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20034] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_response20034.dart
part of openapi.api;

class InlineResponse20034 {
  
  String original = null;
  
  String originalName = null;
  
  String ingredientImage = null;
  
  List<String> meta = [];
  
  List<FoodIngredientsMapProducts> products = [];
  InlineResponse20034();

  @override
  String toString() {
    return 'InlineResponse20034[original=$original, originalName=$originalName, ingredientImage=$ingredientImage, meta=$meta, products=$products, ]';
  }

  InlineResponse20034.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['original'] == null) {
      original = null;
    } else {
          original = json['original'];
    }
    if (json['originalName'] == null) {
      originalName = null;
    } else {
          originalName = json['originalName'];
    }
    if (json['ingredientImage'] == null) {
      ingredientImage = null;
    } else {
          ingredientImage = json['ingredientImage'];
    }
    if (json['meta'] == null) {
      meta = null;
    } else {
      meta = (json['meta'] as List).cast<String>();
    }
    if (json['products'] == null) {
      products = null;
    } else {
      products = FoodIngredientsMapProducts.listFromJson(json['products']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (original != null)
      json['original'] = original;
    if (originalName != null)
      json['originalName'] = originalName;
    if (ingredientImage != null)
      json['ingredientImage'] = ingredientImage;
    if (meta != null)
      json['meta'] = meta;
    if (products != null)
      json['products'] = products;
    return json;
  }

  static List<InlineResponse20034> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20034>() : json.map((value) => new InlineResponse20034.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20034> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20034>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20034.fromJson(value));
    }
    return map;
  }
}

